### PR TITLE
HDDS-4518. Add metrics around Trash Operations.

### PR DESCRIPTION
**Pull request by mukul1987**
## What changes were proposed in this pull request?

This jira adds metrics for Trash Operations

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4518

## How was this patch tested?
Modified an existing test to cover this change.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java [hunk 1]
```diff
@@ -386,6 +393,7 @@ boolean processKeyPath(List<String> keyPathList) {
         OzoneManagerProtocolProtos.OMRequest omRequest =
             getRenameKeyRequest(src, dst);
         try {
+          ozoneManager.getMetrics().incNumTrashFilesRenames();
```
Comment (amaliujia) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java:396]: Out of curiosity: what is the difference between `incNumTrashFilesRenames` and ` incNumTrashRenames`, does the latter contains the former or vice verse?

Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java:396]: `numTrashRenames` would count the renames on the directory while `numTrashFileRenames` would count renames on each file of the directory.

**Review by sadanand48 — COMMENTED** [review]
+1 Looks good to me.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java (hunk 1 above)
Comment (amaliujia) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java:396]: Thanks for the clarification!

**Comment by amaliujia** [comment]
+1 LGTM

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java [hunk 2]
```diff
@@ -1225,6 +1236,15 @@ public void testTrash() throws Exception {
       }
     }, 1000, 180000);
 
+    Assert.assertTrue(getOMMetrics()
+        .getNumTrashDeletes() >= prevNumTrashDeletes);
+    Assert.assertTrue(getOMMetrics()
+        .getNumTrashRenames() > prevNumTrashRenames);
+    Assert.assertTrue(getOMMetrics()
+        .getNumTrashFilesDeletes() >= prevNumTrashFileDeletes);
```
Comment (ayushtkn) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java:1244]: If for `prevNumTrashDeletes` and `prevNumTrashFileDeletes ` the equality hold true, That would mean the metrics didn't increment, which won't serve the purpose of the test. Can we ditch the `=` in the assertion and confirm the metric value gets incremented. Something like -
```
    GenericTestUtils.waitFor(
        () -> getOMMetrics().getNumTrashDeletes() > prevNumTrashDeletes
            && getOMMetrics().getNumTrashFilesDeletes()
            > prevNumTrashFileDeletes, 100, 180000);
```
Unless there is some catch here, in that case it should be ok..

**Review by ayushtkn — COMMENTED** [review]
Thanx @mukul1987 for the work here.
There is a complain from checkstyle which doesn't seems to be introduced here. So should be fine.
Changes LGTM

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java (hunk 2 above)
Comment (mukul1987) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java]: Thanks, Excellent comment. Incorporated it.

**Review by ayushtkn — APPROVED** [review]
+1

**Pull request merged**